kustomize/commands/edit/add/addbuildmetadata.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2022 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package add
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/spf13/cobra"
+	"sigs.k8s.io/kustomize/api/types"
+	"sigs.k8s.io/kustomize/kustomize/v4/commands/internal/kustfile"
+	"sigs.k8s.io/kustomize/kyaml/filesys"
+)
+
+type addBuildMetadataOptions struct {
+	buildMetadataOptions []string
+}
+
+// newCmdAddBuildMetadata adds options to the kustomization's buildMetada field.
+func newCmdAddBuildMetadata(fSys filesys.FileSystem) *cobra.Command {
+	var o addBuildMetadataOptions
+
+	cmd := &cobra.Command{
+		Use:   "buildmetadata",
+		Short: "Adds one or more buildMetadata options to the kustomization.yaml in the current directory",
+		Example: `
+		add buildmetadata {option1},{option2}`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			err := o.Validate(args)
+			if err != nil {
+				return err
+			}
+			return o.RunAddBuildMetadata(fSys)
+		},
+	}
+	return cmd
+}
+
+// Validate validates addBuildMetadata command.
+func (o *addBuildMetadataOptions) Validate(args []string) error {
+	if len(args) != 1 {
```

Comment:
we want to better handle the corner case when user mistakenly put empty string between the buildMetadata (in such case, `len(args) > 1` returns `must specify a buildMetadata option` might be misleading)

How about:
```go
if len(args) == 0 {
   return errors.New("must specify a buildMetadata option")
}
if len(args) > 1 {
    return fmt.Errorf("too many arguments: %s.\nTo provide multiple buildMetadata options, please separate by comma")
}
```

Reply:
done!